Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM openjdk:11-jre-bullseye
+
+RUN mkdir -p bugle-forms-app
+WORKDIR /bugle-forms-app
+
+COPY target/uberjar/bugle-forms-latest-standalone.jar ./app-standalone.jar
```

Comment:
We can simply copy the root dir over and build the jar within the image no? That way, one could simply clone the project, run `docker build .` and have the application up and running. What do you think?

Reply:
I thought about this, and here are the tradeoffs:

Downsides of building inside the container:
- It will lead to a larger image, but this can be mitigated with [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/).
- It will make the build process of the image slower. The consequence: CI will be slower, and afaik we won't be able to cache the lein dependencies anymore in the workflows (not without overly complicating the workflow, at least). This would make deployments slower.

Benefits:
- As you said, easier for the developer to try out the project locally.
- Less coupling between local setup and container image.

I felt that doing it the way I did is an okay tradeoff, because of the assumption that most people trying to locally build this project would likely be doing it for development purposes. In which case, it would be reasonable to assume that they have `lein` installed. In which case, having a local setup to match the deployed image is as easy as
```
lein deps && lein uberjar && docker build .
```

...which does not seem too bad.

Let me know what you think about this.

Reply:
> we won't be able to cache the lein dependencies anymore in the workflows
Ah, that's a good point. Your suggestion makes sense.